src/sif/pedestriancost.cc

```diff
@@ -744,16 +754,32 @@ Cost PedestrianCost::EdgeCost(const baldr::DirectedEdge* edge,
 // Returns the time (in seconds) to make the transition from the predecessor
 Cost PedestrianCost::TransitionCost(const baldr::DirectedEdge* edge,
                                     const baldr::NodeInfo* node,
-                                    const EdgeLabel& pred) const {
+                                    const EdgeLabel& pred,
+                                    const graph_tile_ptr& tile,
+                                    baldr::GraphReader& reader) const {
```

Comment:
really unfortunate to give such power to the costing function, one could do a lot of damage having access to the whole graph inside of costing. ill give a bit of thought if there is another way to get the info we need without passing the graphreader